File: leetcode/118. Pascals Triangle/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/pascals-triangle/

const generate = (numRows: number): number[][] => {
  let pascalTriangle: number[][] = []
  for (let i = 0; i < numRows; i++) {
    let row: number[] = new Array(i + 1)

    row[0] = 1
    row[i] = 1

    for (let j = 1; j < i; j++) {
      row[j] = pascalTriangle[i - 1][j - 1] + pascalTriangle[i - 1][j]
    }

    pascalTriangle.push(row)
  }

  return pascalTriangle
}

console.log(generate(5))

File: leetcode/118. Pascals Triangle/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/pascals-triangle/

class Solution {
  public List<List<Integer>> generate(int numRows) {
    List<List<Integer>> pascalTriangle = new ArrayList<>();
    for (var i = 0; i < numRows; i++) {
      List<Integer> row = new ArrayList<>(i);      
      
      row.add(1);

      for (var j = 1; j < i; j++) {
        int val = pascalTriangle.get(i-1).get(j-1) + pascalTriangle.get(i-1).get(j);
        row.add(val);
      }

      if (i != 0) {
        row.add(1);
      }

      pascalTriangle.add(row);
    }

    return pascalTriangle;
  }
}
